modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountRoleLocalServiceImpl.java

```diff
@@ -174,4 +213,53 @@ public void unassociateUser(
 			new long[] {accountRole.getRoleId()});
 	}
 
+	private DynamicQuery _getDynamicQuery(
+		long accountEntryId, String keywords, OrderByComparator obc) {
+
+		DynamicQuery roleDynamicQuery = roleLocalService.dynamicQuery();
+
+		Disjunction disjunction = RestrictionsFactoryUtil.disjunction();
+
+		disjunction.add(
+			RestrictionsFactoryUtil.ilike(
+				"description", StringUtil.quote(keywords, StringPool.PERCENT)));
+		disjunction.add(
+			RestrictionsFactoryUtil.ilike(
+				"name", StringUtil.quote(keywords, StringPool.PERCENT)));
+
+		roleDynamicQuery.add(disjunction);
+
+		roleDynamicQuery.setProjection(
+			ProjectionFactoryUtil.property("roleId"));
+
+		if (obc != null) {
+			Order order;
+
+			if (obc.isAscending()) {
+				order = OrderFactoryUtil.asc(obc.getOrderByFields()[0]);
+			}
+			else {
+				order = OrderFactoryUtil.desc(obc.getOrderByFields()[0]);
+			}
+
+			roleDynamicQuery.addOrder(order);
+		}
+
+		List<Long> roleIds = roleLocalService.dynamicQuery(roleDynamicQuery);
+
+		if (roleIds.isEmpty()) {
+			return null;
+		}
+
+		DynamicQuery accountRoleDynamicQuery =
+			accountRoleLocalService.dynamicQuery();
+
+		accountRoleDynamicQuery.add(
+			RestrictionsFactoryUtil.eq("accountEntryId", accountEntryId));
+		accountRoleDynamicQuery.add(
+			RestrictionsFactoryUtil.in("roleId", roleIds));
```

Comment:
I can't pass in the subquery directly here because the Role class is in the portal classloader and can't be executed in the module classpath. When I tried it, I got the `org.hibernate.MappingException: Unknown entity` exception.

After some research, looks like it's a limitation that it's not possible to do a dynamic query on two different tables on different class loaders. Ref: https://issues.liferay.com/browse/LPS-26849